### DISCUSSION TITLE
How to interpolatively resize down using bilinear sampling

### DISCUSSION THREAD

**Discussion by thegreatgramcracker** [dlemstra/Magick.NET#1583] [Help] [answered]
Hello. I am trying to resize an image to 1/8 of it's resolution using bilinear resizing, but the code I expect to work doesn't seem to give the correct results.
```cs
MagickImage magickImage = new MagickImage(decompressedBytes);
magickImage.Settings.SetDefine(MagickFormat.Dds, "compression", "none");

magickImage.InterpolativeResize(magickImage.Width / 8, magickImage.Height / 8, PixelInterpolateMethod.Bilinear);
```
The result looks something like this (I upscaled to original size using nearest neighbor):
![m02_gate_wall_top](https://github.com/dlemstra/Magick.NET/assets/40072952/3701b370-47fd-4ee2-9791-577a8983eca5)

To me it doesn't appear to be resized using bilinear interpolation. Doing the same thing in Paint.NET looks like this:
![m02_gate_wall_top](https://github.com/dlemstra/Magick.NET/assets/40072952/2007c834-8c8e-4137-9c1a-611c71e3752a)

I might be mistaken though because I'm not sure if there is more interpolation going on under the hood in Paint.NET, but if so then I am hoping to find out more how I can achieve a result similar to Paint.NET.

Any help or feedback would be greatly appreciated!

**Comment by dlemstra** [comment]
Can you share your input image? It looks like Paint.NET is just doing a normal resize?

**Reply by thegreatgramcracker** [reply]
I'm not able to attach a dds it seems but here is a png version of the original image. I actually managed to get a better result by just doing Resize() instead of InterpolativeResize(), and I added `magickImage.Settings.SetDefine("filter", "sinc");` before I resized, which I'm not sure if I really set it right and if it made a difference. I'm fairly sure Paint.NET just did a normal bilinear interpolated resize, but my code in magick.NET didn't seem to use bilinear interpolation when I specified. Does InterpolativeResize() only work for upscaling?

![m02_gate_wall_top](https://github.com/dlemstra/Magick.NET/assets/40072952/82c211dc-111d-4b43-bb91-65d56d6347a1)

**Reply by dlemstra** [reply, marked as answer]
You should use `image.FilterType` instead of that define. You can find more information about what `InterpolativeResize` does here: https://imagemagick.org/Usage/resize/#interpolative-resize.

**Reply by thegreatgramcracker** [reply]
Thank you, the link you provided gave me some helpful advice.
